File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {

        map<int, int> lCounterMap;
        vector<int> lRes;

        for (const int& eachNum : nums) {
            if (lCounterMap.count(eachNum)) {
                lRes.push_back(eachNum);
            }
            lCounterMap[eachNum]++;
        }

        return lRes;
    }
};

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:

        hashMap = {}
        res = []

        for each in nums:
            if each not in hashMap:
                hashMap[each] = 1
            else:
                hashMap[each] += 1
                if hashMap[each] > 1:
                    res.append(each)
        
        return res